packages/flagging/src/precomputeClient.ts

```diff
@@ -0,0 +1,183 @@
+      },
+    },
+    precomputedFlagStore: memoryOnlyPrecomputedFlagsStore,
+  })
+
+  public static initialized = false
+
+  private subject: Subject
+  private precomputedFlagStore: ISyncStore<PrecomputedFlag>
+
+  public constructor({
+    subject,
+    precomputedFlagStore,
+  }: {
+    subject: Subject
+    precomputedFlagStore: ISyncStore<PrecomputedFlag>
+  }) {
+    this.subject = subject
+    this.precomputedFlagStore = precomputedFlagStore
+  }
+
+  public getStringAssignment(flagKey: string, defaultValue: string): string {
+    return this.getPrecomputedAssignment(flagKey, defaultValue, VariationType.STRING)
+  }
+
+  public getBooleanAssignment(flagKey: string, defaultValue: boolean): boolean {
+    return this.getPrecomputedAssignment(flagKey, defaultValue, VariationType.BOOLEAN)
+  }
+
+  public getIntegerAssignment(flagKey: string, defaultValue: number): number {
+    return this.getPrecomputedAssignment(flagKey, defaultValue, VariationType.INTEGER)
+  }
+
+  public getNumericAssignment(flagKey: string, defaultValue: number): number {
+    return this.getPrecomputedAssignment(flagKey, defaultValue, VariationType.NUMERIC)
+  }
+
+  public getJSONAssignment(flagKey: string, defaultValue: object): object {
+    return this.getPrecomputedAssignment(flagKey, defaultValue, VariationType.JSON, (value) => {
+      if (typeof value === 'string') {
+        try {
+          const parsed = JSON.parse(value)
+          return typeof parsed === 'object' && parsed !== null ? parsed : defaultValue
+        } catch {
+          return defaultValue
+        }
+      }
+      return typeof value === 'object' && value !== null ? value : defaultValue
+    })
+  }
+
+  private getPrecomputedAssignment<T>(
+    flagKey: string,
+    defaultValue: T,
+    _expectedType: VariationType,
+    valueTransformer: (value: unknown) => T = (v) => v as T
+  ): T {
+    // validateNotBlank(flagKey, 'Invalid argument: flagKey cannot be blank')
+
+    const precomputedFlag = this.getPrecomputedFlag(flagKey)
+
+    if (precomputedFlag === null) {
+      return defaultValue
+    }
+
+    const result: FlagEvaluationWithoutDetails = {
+      flagKey,
+      format: 'PRECOMPUTED',
+      subjectKey: this.subject.key ?? '',
+      subjectAttributes: {
+        ...this.subject.attributes?.numericAttributes,
+        ...this.subject.attributes?.categoricalAttributes,
+      },
+      variation: {
+        key: precomputedFlag.variationKey ?? '',
+        value: precomputedFlag.variationValue,
+      },
+      allocationKey: precomputedFlag.allocationKey ?? '',
+      extraLogging: precomputedFlag.extraLogging ?? {},
+      doLog: precomputedFlag.doLog,
+      entityId: null,
+    }
+
+    try {
+      const transformedValue =
+        result.variation?.value !== undefined ? valueTransformer(result.variation.value) : defaultValue
+      // TODO: add logging or open feature hook
+      // if (result?.doLog) {
+      //   this.logAssignment(result)
+      // }
```

Comment:
👍 good to be thinking about this.
I imagine int he future the logger will also be passed the allocation type, so finer grained logging or not could be controlled by client vs. doLog. I can see a world where we rid ourselves of doLog entirely.